File: LeetCode-2023/1828.统计一个圆中点的数目.cpp
/*
 * @lc app=leetcode.cn id=1828 lang=cpp
 *
 * [1828] 统计一个圆中点的数目
 */

// @lc code=start
class Solution {
public:
    vector<int> countPoints(vector<vector<int>>& points, vector<vector<int>>& queries) {
        int m = points.size();
        int n = queries.size();
        vector<int> ans(n);
        for(int i = 0; i < n; i++) {
            int cx = queries[i][0];
            int cy = queries[i][1];
            int cr = queries[i][2];
            for(int j = 0; j < m; j++) {
                int px = points[j][0];
                int py = points[j][1];
                if((cx - px) * (cx - px) + (cy - py) * (cy - py) <= cr * cr) {
                    ans[i]++;
                }
            }
        }
        return ans;
    }
};
// @lc code=end

File: LeetCode-2023/1828.统计一个圆中点的数目.py
#
# @lc app=leetcode.cn id=1828 lang=python3
#
# [1828] 统计一个圆中点的数目
#

# @lc code=start
class Solution:
    def countPoints(self, points: List[List[int]], queries: List[List[int]]) -> List[int]:
        ans = [0] * len(queries)
        for i, (cx, cy, cr) in enumerate(queries):
            for (px, py) in points:
                if (cx - px) ** 2 + (cy - py) ** 2 <= cr ** 2 :
                    ans[i] += 1
        return ans
# @lc code=end

